ledger/ledger-api-integration-tests/src/test/itsuite/scala/com/digitalasset/platform/tests/integration/ledger/api/ContractKeysIT.scala

```diff
@@ -0,0 +1,465 @@
+          testIdsGenerator.testCommandId("CK-test-bobTKO"),
+          Bob,
+          templateIds.textKeyOperations,
+          Record(fields = List(RecordField(value = Bob.asParty)))
+        )
+
+        // unauthorized lookups are not OK
+        // both existing lookups...
+        lookupNone = Value(Value.Sum.Optional(Optional(None)))
+        lookupSome = (cid: String) => Value(Value.Sum.Optional(Optional(Some(cid.asContractId))))
+        _ <- ctx.testingHelpers.failingExercise(
+          testIdsGenerator.testCommandId("CK-test-bob-unauthorized-1"),
+          Bob,
+          templateIds.textKeyOperations,
+          bobTKO.contractId,
+          "TKOLookup",
+          Value(
+            Value.Sum.Record(
+              Record(fields = List(
+                RecordField(value = textKeyKey(Alice, key)),
+                RecordField(value = lookupSome(cid1.contractId)))))),
+          Code.INVALID_ARGUMENT,
+          if (languageVersion precedes LanguageVersion.checkSubmitterInMaintainers) {
+            "requires authorizers"
+          } else {
+            "Expected the submitter 'Bob' to be in maintainers 'Alice'"
+          }
+        )
+        // ..and non-existing ones
+        _ <- ctx.testingHelpers.failingExercise(
+          testIdsGenerator.testCommandId("CK-test-bob-unauthorized-2"),
+          Bob,
+          templateIds.textKeyOperations,
+          bobTKO.contractId,
+          "TKOLookup",
+          Value(
+            Value.Sum.Record(
+              Record(fields = List(
+                RecordField(value = textKeyKey(Alice, "bogus-key")),
+                RecordField(value = lookupNone))))),
+          Code.INVALID_ARGUMENT,
+          if (languageVersion precedes LanguageVersion.checkSubmitterInMaintainers) {
+            "requires authorizers"
+          } else {
+            "Expected the submitter 'Bob' to be in maintainers 'Alice'"
+          }
+        )
+        // successful, authorized lookup
+        _ <- ctx.testingHelpers.simpleExercise(
+          testIdsGenerator.testCommandId("CK-test-alice-lookup-found"),
+          Alice,
+          templateIds.textKeyOperations,
+          aliceTKO.contractId,
+          "TKOLookup",
+          Value(
+            Value.Sum.Record(
+              Record(fields = List(
+                RecordField(value = textKeyKey(Alice, key)),
+                RecordField(value = lookupSome(cid1.contractId))))))
+        )
+        // successful fetch
+        _ <- ctx.testingHelpers.simpleExercise(
+          testIdsGenerator.testCommandId("CK-test-alice-fetch-found"),
+          Alice,
+          templateIds.textKeyOperations,
+          aliceTKO.contractId,
+          "TKOFetch",
+          Value(
+            Value.Sum.Record(
+              Record(fields = List(
+                RecordField(value = textKeyKey(Alice, key)),
+                RecordField(value = cid1.contractId.asContractId)))))
+        )
+        // failing, authorized lookup
```

Comment:
This line says failing but the following is not a `failingExercise`, am I missing something?

Reply:
Addressed in 9bf50c0.